src/main/java/org/radarcns/domain/restapi/Source.java

```diff
@@ -0,0 +1,152 @@
+package org.radarcns.domain.restapi;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.radarcns.domain.restapi.header.EffectiveTimeFrame;
+
+public class Source {
+
+    @JsonProperty
+    private String sourceId;
+
+    @JsonProperty
+    private String sourceName;
+
+    @JsonProperty
+    private String sourceTypeProducer;
+
+    @JsonProperty
+    private String sourceTypeModel;
+
+    @JsonProperty
+    private String sourceTypeCatalogVersion;
+
+    @JsonProperty
+    private Boolean assigned;
+
+    @JsonProperty
+    private String status;
+
+    @JsonProperty
+    private EffectiveTimeFrame effectiveTimeFrame;
+
+    /**
+     * Default constructor.  Note that this does not initialize fields
+     * to their default values from the schema.  If that is desired then
+     * one should use <code>newBuilder()</code>.
+     */
+    public Source() {}
+
+    /**
+     * All-args constructor.
+     * @param id Source identifier.
+     * @param type Source name.
+     */
+    public Source(java.lang.String id, java.lang.String type) {
+        this.sourceId = id;
+        this.sourceTypeModel = type;
+//        this.summary = summary;
+    }
+    public Source sourceId(String sourceId) {
+        this.sourceId = sourceId;
+        return this;
+    }
+
+    public String getSourceId() {
+        return sourceId;
+    }
+
+    public void setSourceId(String sourceId) {
+        this.sourceId = sourceId;
+    }
+
+    public Source sourceName(String sourceName) {
```

Comment:
I'm not sure about this, I feel like we're mixing paradigms here (builder and pojo pattern) Having both a setter and a hybrid builder seems superfluous.